extensions/interactions/NumberWithUnits/directives/NumberWithUnits.js

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+oppia.directive('oppiaInteractiveNumberWithUnits', [
+  'HtmlEscaperService', 'UrlInterpolationService',
+  function(HtmlEscaperService, UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {
+        onSubmit: '&',
+        // This should be called whenever the answer changes.
+        setAnswerValidity: '&'
+      },
+      templateUrl: UrlInterpolationService.getExtensionResourceUrl(
+        '/interactions/NumberWithUnits/directives/' +
+        'number_with_units_interaction_directive.html'),
+      controller: [
+        '$scope', '$attrs', 'FocusManagerService',
```

Comment:
Is there any reason to import this?

Reply:
Yes, we do need these services. It helps in setting the getting the focus label of the input and used in other interactions like `FractionInput` as well.

Reply:
The reason I asked is because this controller is not using this service and the service I mentioned in comment below. So maybe you missed to write that code?

Reply:
Yes, they are not used here. I'll remove them.

Reply:
Done!